src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -234,24 +236,43 @@ private void loadSchematic()
 
         try
         {
-            job.setSchematic(new SchematicWrapper(world, workOrder.getSchematicName()));
+            job.setStructure(new StructureWrapper(world, workOrder.getStructureName()));
         }
-        catch (IllegalStateException e)
+        catch (RuntimeException e)
         {
-            Log.getLogger().warn(String.format("Schematic: (%s) does not exist - removing build request", workOrder.getSchematicName()), e);
-            job.setSchematic(null);
+            Log.getLogger().warn(String.format("StructureProxy: (%s) does not exist - removing build request", workOrder.getStructureName()), e);
+            job.setStructure(null);
             return;
         }
 
-        job.getSchematic().rotate(workOrder.getRotation());
-        job.getSchematic().setPosition(pos);
+        Colony colony = worker.getColony();
+        if(workOrder instanceof WorkOrderBuildDecoration)
+        {
+            job.getStructure().rotate(workOrder.getRotation());
+        }
+        else if(colony != null)
+        {
+            if(workOrder.getRotation() == 0)
+            {
+                IBlockState blockState = world.getBlockState(pos);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'blockState' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)